docs/proposals/scheduling/20230707-eviction-arbitrate-mechanism-in-descheduler.md

```diff
@@ -0,0 +1,256 @@
+// can implement different SortFn according to their needs.
+type SortFn func(jobs []*v1alpha1.PodMigrationJob, client client.Client) []*v1alpha1.PodMigrationJob
+~~~
+
+###### GroupFilter PodMigrationJob
+
+![image](/docs/images/arbitration-mechanism-groupfilter-design.svg)
+
+Aggregate PodMigrationJob according to different workloads and filter them based on different strategies.
+
+- According to Workload
+  - Group: Aggregate PodMigrationJob by workload.
+  - Filter: 
+    - Check how many PodMigrationJob of each workload are in the Running state, and record them as ***migratingReplicas***. If the **migratingReplicas** reach a certain threshold, excess parts will be excluded. 
+    - Check the number of **unavailableReplicas** of each workload, and determine whether the **unavailableReplicas** exceeds the **MaxUnavailablePerWorkload**, exclude excess parts.
+- According to Node
+  - Group: Aggregate PodMigrationJob by Node.
+  - Filter:
+    - Check the number of Pods being migrated on the node where each target Pod is located. If it exceeds the maximum migration amount for a single node, exclude excess parts.
+- According to Namespace
+  - Group: Aggregate PodMigrationJob by Namespace.
+  - Filter: 
+    - Check the number of Pods being migrated in the Namespace where each target Pod is located. If it exceeds the maximum migration amount for a single Namespace, exclude excess parts
+
+The definition of the type `GroupFilterFn` is as follows.
+
+~~~ go
+type GroupFilterFn func(jobs []*v1alpha1.PodMigrationJob, client client.Client) []*v1alpha1.PodMigrationJob
+~~~
+
+###### Select PodMigrationJob
+
+![image](/docs/images/arbitration-mechanism-select-design.svg)
+
+- Sort PodMigrationJob slices based on the map.
+- Place the first **n** elements of slice to the rate limit queue.
+- Remove these elements from the arbitration collection.
+
+#### Controller Configuration
+
+User can configure the `MigrationControllerArgs.ArbitrationArgs` through Koordinator Descheduler ConfigMap. 
+
+```go
+// MigrationControllerArgs holds arguments used to configure the MigrationController
+type MigrationControllerArgs struct {
+	...
+	
+    // Arbitration define the control parameters of the arbitration mechanism.
+    // +optional
+    Arbitration *ArbitrationArgs
+}
+
+// ArbitrationArgs holds arguments used to configure the Arbitration Mechanism.
+type ArbitrationArgs struct {
+    // Enabled defines if Arbitration Mechanism should be enabled.
+    // Default is true
+    Enabled bool
+    
+    // Interval defines the running interval of the Arbitration Mechanism.
+    // Default is 1 minute
+    Interval time.Duration
+    
+    // ProcessQueueQPS defines the speed of process queues (per second)
+    // Default to 3
+    ProcessQueueQPS int
+    
+    // JobSelectRatio defines the proportion of the number of PodMigrationJobs 
+    // that the Arbitration Mechanism passes each time.
+    // Default to 1.2
+    JobSelectRatio float32
```

Comment:
Don't quite understand what `JobSelectRatio` does, please add some comments.

Reply:
I used to consider putting partial PodMigrationJobs passed the Arbitration Mechanism into the workqueue. Now it's useless.